lib/rules/no-whitespace-before-property.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @fileoverview Rule to disallow whitespace before properties
+ * @author Kai Cataldo
+ * @copyright 2015 Kai Cataldo. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+"use strict";
+
+var astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var sourceCode = context.getSourceCode();
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    return {
+        MemberExpression: function(node) {
+            var obj = node.object;
+            var prop = node.property;
+
+            if (astUtils.isTokenOnSameLine(obj, prop)) {
```

Comment:
Better to use the `SourceCode` version of this method, which checks for comments as well.